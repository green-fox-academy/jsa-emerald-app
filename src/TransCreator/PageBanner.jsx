import React from 'react';
import {
  View, Text,
} from 'react-native';
import { Icon } from 'react-native-elements';
import { LinearGradient } from 'expo-linear-gradient';

import PropTypes from 'prop-types';
import themeStyle from '../Common/themeStyle';
import themeColor from '../Common/Color';
import styles from './Style';

export default function PageBanner({
  transLabel, transType, transAmount, transExpression,
}) {
  let displayAmount = transAmount === '' ? '0.00' : transAmount;
  displayAmount = transType === 'Expense' ? `-$${displayAmount}` : `+$${displayAmount}`;
  return (
    transLabel.name
      ? (
        <LinearGradient
          colors={transType === 'Expense' ? themeColor.gradientColor.red : themeColor.gradientColor.green}
          start={[0.1, 0.9]}
          end={[0.9, 0.1]}
        >
          <View style={[themeStyle.headerFormat, styles.headerFormat]}>
            <View style={styles.headerContainer}>
              <Icon name={transLabel.icon} type={transLabel.iconFamily} color="#ffffff" size={40} />
              <Text style={[styles.headerText, { color: '#ffffff' }]}>{transLabel.name || 'undefined'}</Text>
            </View>
            <View style={{
              flex: 1, flexDirection: 'column', alignItems: 'flex-end',
            }}
            >
              <Text style={{ color: '#ffffff', fontSize: 20, fontWeight: '400' }}>{ displayAmount }</Text>
              <Text style={{ color: '#EEEEEE', fontSize: 17, fontWeight: '400' }}>{transExpression}</Text>
            </View>
          </View>
        </LinearGradient>
      )
      : (
        <View style={[themeStyle.headerFormat, styles.headerFormat]}>
          <Text style={styles.headerText}>Select a category</Text>
        </View>
      )
  );
}

PageBanner.propTypes = {
  transLabel: PropTypes.shape({
    name: PropTypes.string,
    icon: PropTypes.string,
    iconFamily: PropTypes.string,
    color: PropTypes.string,
  }),
  transType: PropTypes.string,
  transAmount: PropTypes.string,
  transExpression: PropTypes.string,
};

PageBanner.defaultProps = {
  transLabel: {
    name: '',
    icon: '',
    iconFamily: '',
    color: '',
  },
  transType: '',
  transAmount: '',
  transExpression: '',
};
