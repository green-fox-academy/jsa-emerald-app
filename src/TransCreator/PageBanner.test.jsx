import React from 'react';
import renderer from 'react-test-renderer';
import PageBanner from './PageBanner';

describe('<PageBanner/>', () => {
  it('render case 1', () => {
    const tree = renderer.create(<PageBanner
      transLabel={{
        name: '',
        icon: 'home',
        iconFamily: 'antdesign',
        color: '#08addb',
      }}
      transType="Expense"
      transAmount=""
      expStr=""
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render case 2', () => {
    const tree = renderer.create(<PageBanner
      transLabel={{
        name: 'House',
        icon: 'home',
        iconFamily: 'antdesign',
        color: '#08addb',
      }}
      transType="Income"
      transAmount="12.10"
      expStr=""
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render case 3', () => {
    const tree = renderer.create(<PageBanner
      transLabel={{
        name: 'House',
        icon: 'home',
        iconFamily: 'antdesign',
        color: '',
      }}
      transType="Expense"
      transAmount="12.10"
      expStr="#08addb"
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
