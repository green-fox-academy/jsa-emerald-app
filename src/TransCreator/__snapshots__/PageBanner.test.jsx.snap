// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageBanner/> render case 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
      },
      Object {
        "height": 140,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "400",
      }
    }
  >
    Select a category
  </Text>
</View>
`;

exports[`<PageBanner/> render case 2 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4281321625,
        4286115731,
      ],
      "endPoint": Array [
        0.9,
        0.1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0.1,
        0.9,
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
        },
        Object {
          "height": 140,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        House
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        $12.10
      </Text>
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 17,
            "fontWeight": "400",
          }
        }
      >
        
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`<PageBanner/> render case 3 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4293488030,
        4294945423,
      ],
      "endPoint": Array [
        0.9,
        0.1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0.1,
        0.9,
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
        },
        Object {
          "height": 140,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        House
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        -$12.10
      </Text>
      <Text
        style={
          Object {
            "color": "#EEEEEE",
            "fontSize": 17,
            "fontWeight": "400",
          }
        }
      >
        #08addb
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
